lib/vtls/bearssl.c

```diff
@@ -474,6 +474,18 @@ static CURLcode bearssl_connect_step1(struct Curl_easy *data,
     hostname = snihost;
   }
 
+  /* give application a chance to interfere with SSL set up. */
+  if(data->set.ssl.fsslctx) {
+    Curl_set_in_callback(data, true);
+    ret = (*data->set.ssl.fsslctx)(data, &backend->ctx,
+                                   data->set.ssl.fsslctxp);
+    Curl_set_in_callback(data, false);
+    if(ret) {
+      failf(data, "BearSSL: error signaled by ssl ctx callback");
+      return ret;
+    }
+  }
+
   if(!br_ssl_client_reset(&backend->ctx, hostname, 0))
```

Comment:
shouldn't the ctx callback call come after the reset ctx call?

Reply:
`br_ssl_client_reset` is documented as "Prepare or reset a client context for a new connection", it doesn't completely wipe the context (that's `br_ssl_client_zero`) or reinitialize the settings that may have been set by the callback. So this is the right place for it.